javaslang/src/main/java/javaslang/collection/JavaConverters.java

```
@@ -0,0 +1,437 @@
+            return new ListView<>(getDelegate().subSequence(fromIndex, toIndex), isMutable());
+        }
+
+        @Override
+        public Object[] toArray() {
+            return getDelegate().toJavaArray();
+        }
+
+        @Override
+        public <T1> T1[] toArray(T1[] array) {
+            return getDelegate().toJavaList().toArray(array);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            } else if (o instanceof java.util.List) {
+                return Collections.areEqual(getDelegate(), (java.util.List<?>) o);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return getDelegate().hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return getDelegate().mkString("[", ", ", "]");
+        }
+
+        // -- private helpers
+
+        private boolean setDelegateAndCheckChanged(Seq<T> delegate) {
+            // TODO: or `return previousDelegate != delegate`?
+            return setDelegate(delegate).size() != delegate.size();
+        }
+
+        private T setDelegateAndGetPreviousElement(int index, Seq<T> delegate) {
+            return setDelegate(delegate).get(index);
+        }
+
+        private static class ListIterator<T> extends HasDelegate<IndexedSeq<T>> implements java.util.ListIterator<T> {
+
+            private static final long serialVersionUID = 1L;
+
+            private int index;
+
+            static <T> ListIterator<T> of(Seq<T> seq, int index) {
+                final IndexedSeq<T> indexedSeq = (seq instanceof IndexedSeq) ? (IndexedSeq<T>) seq : seq.toVector();
+                return new ListIterator<>(indexedSeq, index);
+            }
+
+            ListIterator(IndexedSeq<T> delegate, int index) {
+                super(delegate, true); // TODO: always mutable?
```

You know what I'm going to say to that question. :-)

Exceptions aren't nice, but silent failures are worse than noisy failures. Pretending an immutable collection is mutable is bound to end in tears. You would be quietly mutating a secret copy of the immutable list. I'm not sure any other code would ever see that mutated copy.